scripts/check-docs/src/main.rs

```diff
@@ -0,0 +1,122 @@
+extern crate core;
+
+use std::collections::{HashMap, HashSet};
+use std::env;
+use std::path::Path;
+
+fn main() {
+    let mut stack: Vec<&str> = vec![];
+    let mut valid_anchors: HashMap<&str, HashSet<&str>> = HashMap::new();
+
+    let rg_project = std::process::Command::new("grep")
+        .args(["-I", "-H", "-R", "--exclude-dir=scripts", "ANCHOR", "."])
```

Comment:
Two points:
1. (more important) consider all environments in which you expect this script to work. For example, on macs a different grep is shipped (BSD version) than on most linux disros (gnu grep) and they don't have the same flags. The only thing you can be certain they both have is the posix [subset of features](https://pubs.opengroup.org/onlinepubs/009695399/utilities/grep.html).

2. Prefer writing the long versions of flags in scripts to increase readability.

Reply:
1. They seem to be the same. At least the ones we use.
2. I think the readability is the same. :shrug:

Reply:
"-I -H -R" is as readable as "--binary-files=without-match --with-filename --dereference-recursive" ?